protobuf-lite/src/main/java/io/grpc/protobuf/lite/ProtoLiteUtils.java

```diff
@@ -107,23 +110,38 @@ public T parse(InputStream stream) {
             }
           }
         }
+        CodedInputStream cis;
+        if (stream instanceof ReadableBuffer) {
```

Comment:
I was originally thinking the cutoff could be as high as ~1M. I could live with 4 MB, and that is slightly interesting as it coincides with our default maximum message size.

I do think we should remember that Android uses this same code path, though. Although Android would be much less likely to have such large of messages.